hyperspy/tests/utils/test_roi.py

```diff
@@ -308,10 +308,39 @@ def test_out(self):
         d = s.data.sum()
         sr.data += 2
         nt.assert_equal(d + sr.data.size * 2, s.data.sum())
+        r.left += 2
+        r.right += 2
```

Comment:
And corresponding ones for position (`x` and `y`?). Case in point: This triggers a ton more events than it needs to (first move and resize for both ROI and any connected widgets, then a resize of ROI and widgets) and any widgets will moved staggered. It will also fail if you try to move the left edge past the right one (e.g. if these lines said `+= 10`).